### PR TITLE
Update Stacks.js overview.md

### PR DESCRIPTION
**Pull request by mefrem**
Kicking off the Overview overhaul for the Stacks.js portion of our docs.

### Description

- Condensed introductory language to four lines
- Explicitly mention common Stacks.js use cases—authenticating users and using post-conditions—and link to guides
- Point developers immediately to Stacks.js-starter-templates, which demonstrate Stacks.js code for inspection

#### Issues involved
- https://github.com/hirosystems/stacks.js/issues/1556
- Overlaps with typo change on same page https://github.com/hirosystems/stacks.js/pull/1564

### Checklist

- [ ] Unit tested updated code paths
- [ ] Tagged 1 of @janniks or @zone117x for review

### CODE REVIEW

File: docs/overview.md [hunk 1]
```diff
@@ -24,22 +30,27 @@ Users may use the Stacks.js CLI directly to send a transaction.
...
+--- 
 
-There are three main integrations used by Stacks enabled applications:
+While there are many Stacks.js use cases, we find that there are three main integrations used by Stacks-enabled applications:
 
 <!-- todo: add a card and better how-to-guide for each, (eg add message signing backend checking) -->
 
 - **Authentication**: Register and sign users in with identities on the Stacks blockchain
 - **Transaction signing**: Prompt users to sign and broadcast transactions to the Stacks blockchain
 - **Data storage**: Save and retrieve data for users with [Gaia](https://docs.stacks.co/build-apps/references/gaia)
```
Comment (ryanwaits) [inline docs/overview.md]: i would almost remove the callout to Gaia here

Comment (ryanwaits) [reply docs/overview.md]: and maybe we add reading state from a smart contract (calling smart contract functions)

**Review by ryanwaits — COMMENTED** [review]
this is a lot cleaner already - made a few small comments but otherwise looks good

**Comment by mefrem** [comment]
Should pull Micaiah's PR; replace Stacks.js/connect library link with https://connect.stacks.js.org/

File: docs/overview.md (hunk 1 above)
Comment (mefrem) [reply docs/overview.md]: Agreed. Updated with https://github.com/hirosystems/stacks.js/pull/1568/commits/f550d16d0169666d34d8555385e140c732c7d08e

**Comment by janniks** [comment]
Love this, let's sync today -- I also made some changes. 🙌

**Pull request closed without merging**